src/schema/document/helpers.rs

```diff
@@ -0,0 +1,208 @@
+//! Implementations of some of the core traits on varius types to improve the ergonomics
```

Comment:
I'd rename it from `helpers.rs` to something more concrete like`serde_json_impl.rs` or similar